zaza/openstack/charm_tests/rabbitmq_server/tests.py

```diff
@@ -356,25 +370,48 @@ def test_921_remove_unit(self):
         zaza.model.set_application_config('rabbitmq-server', config)
         rmq_utils.wait_for_cluster()
 
-        units = zaza.model.get_units(self.application_name)
-        removed_unit = units[-1]
-        left_units = units[:-1]
+        all_units = zaza.model.get_units(self.application_name)
+        removed_unit = all_units[-1]
+        left_units = all_units[:-1]
 
+        logging.info('Simulating unit {} removal'.format(removed_unit))
         zaza.model.run_on_unit(removed_unit.entity_id, 'hooks/stop')
+        logging.info('Waiting until unit {} reaches "waiting" state'
+                     ''.format(removed_unit))
         zaza.model.block_until_unit_wl_status(removed_unit.entity_id,
                                               "waiting")
 
-        unit_host_names = generic_utils.get_unit_hostnames(left_units)
-        unit_node_names = []
-        for unit in unit_host_names:
-            unit_node_names.append('rabbit@{}'.format(unit_host_names[unit]))
-        errors = []
-
-        for u in left_units:
-            e = self._retry_check_unit_cluster_nodes(u,
-                                                     unit_node_names)
-            if e:
-                errors.append(e)
+        def check_units(units):
+            unit_host_names = generic_utils.get_unit_hostnames(units)
+            unit_node_names = []
+            for unit in unit_host_names:
+                unit_node_names.append('rabbit@{}'.format(
+                    unit_host_names[unit]))
+            errors = []
+
+            for u in units:
+                e = self._retry_check_unit_cluster_nodes(u,
+                                                         unit_node_names)
+                if e:
+                    errors.append(e)
+
+            self.assertFalse(errors, msg=errors)
+
+        logging.info('Checking that all units except for {} are present'
+                     'in the cluster'.format(removed_unit))
```

Comment:
I think you should use `format(removed_unit.entity_id)` to have a more human-readable output